### PR TITLE
Allow for coercion when inserting into table

### PR DESCRIPTION
**Pull request by sopel39**

### CODE REVIEW

**Comment by sopel39** [comment]
FYI: @arhimondr @martint

File: presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java [hunk 1]
```diff
@@ -605,13 +610,20 @@ private static boolean typesMatchForInsert(Iterable<Type> tableTypes, Iterable<T
...
                 }
-                if (!typesMatchForInsert(tableType.getTypeParameters(), queryType.getTypeParameters())) {
+                if (!typesMatchForInsert(tableType.getTypeParameters(), queryType.getTypeParameters(), true)) {
                     return false;
                 }
             }
             else {
-                if (!Objects.equals(tableType, queryType) && !isTypeOnlyCoercion(queryType.getTypeSignature(), tableType.getTypeSignature())) {
-                    return false;
+                if (structuredType) {
```
Comment (arhimondr) [inline presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java]: Why do we have 2 different paths for primitive and structured types? Can't it be just one single path? I think it should be possible to insert `ARRAY[INTEGER]` to `ARRAY[DOUBLE]` column. As i know only `MAP` cast operator is not specified.

**Comment by arhimondr** [comment]
lgtm % comment

File: presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java [hunk 2]
```diff
@@ -588,6 +588,11 @@ protected RelationType visitInsert(Insert insert, AnalysisContext context)
 
     private static boolean typesMatchForInsert(Iterable<Type> tableTypes, Iterable<Type> queryTypes)
     {
+        return typesMatchForInsert(tableTypes, queryTypes, false);
+    }
+
+    private static boolean typesMatchForInsert(Iterable<Type> tableTypes, Iterable<Type> queryTypes, boolean structuredType)
```
Comment (losipiuk) [inline presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java]: s/structuredType/structuralType/

**Comment by sopel39** [comment]
applied comments

**Comment by martint** [comment]
Merged, thanks!

**Pull request closed without merging**